pkg/policy/api/selector.go

```diff
@@ -234,6 +234,8 @@ var (
 		labels.IDNameHost:       newReservedEndpointSelector(labels.IDNameHost),
 		labels.IDNameRemoteNode: newReservedEndpointSelector(labels.IDNameRemoteNode),
 		labels.IDNameWorld:      newReservedEndpointSelector(labels.IDNameWorld),
+		labels.IDNameWorldIPv4:  newReservedEndpointSelector(labels.IDNameWorldIPv4),
+		labels.IDNameWorldIPv6:  newReservedEndpointSelector(labels.IDNameWorldIPv6),
```

Comment:
Shouldn't the world reserved selector also select IPv4 and IPv6 world labels now?

The issue IIRC is that a `0.0.0.0/0` CIDR policy was implicitly translated into a world selector, but it should really be translated to world-ipv4 selector. However, if a user actually adds `ToEntities: - world` then that should select both IPv4 and IPv6 destinations, no?

Reply:
I guess the answer to this question depends on whether the new reserved identities are declared with the world label in addition to the world-$proto or not.

Reply:
(If we do change this, it looks like it needs to happen here and in `entity.go`.)

Reply:
Making those selectable is out-of-scope for this PR. That's something we can certainly explore in the future.

Reply:
(same thread as https://github.com/cilium/cilium/pull/22625#discussion_r1293797898, resolving per that comment)